pkgs/development/python-modules/marqo/default.nix

```
@@ -0,0 +1,62 @@
+{ lib, buildPythonPackage, fetchFromGitHub, pytestCheckHook, setuptools
+, requests, packaging, pip, urllib3, pydantic, requests-mock, numpy, pillow
+, dataclasses-json }:
+
+buildPythonPackage rec {
+  name = "marqo";
+  version = "3.9.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "marqo-ai";
+    repo = "py-marqo";
+    rev = "refs/tags/${version}";
+    hash = "sha256-X7qUXdLvI2U0nNcWvB2cmZcnVAzOQc2UiX1mANUZSe4=";
+  };
+
+  build-system = [ setuptools ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  disabledTestPaths = [
+    # Tests require network
+    "tests/v2_tests/test_tensor_search.py"
+    "tests/v2_tests/test_client.py"
+    "tests/v2_tests/test_get_settings.py"
+    "tests/v2_tests/test_tensor_search.py"
+    "tests/v2_tests/test_add_documents.py"
+    "tests/v2_tests/test_delete_documents.py"
+    "tests/v2_tests/test_demos.py"
+    "tests/v2_tests/test_custom_vector_search.py"
+    "tests/v2_tests/test_create_index.py"
+    "tests/v2_tests/test_image_chunking.py"
+    "tests/v2_tests/test_telemetry.py"
+    "tests/v2_tests/test_score_modifier_search.py"
+    "tests/v2_tests/test_model_cache_management.py"
+    "tests/v2_tests/test_embed.py"
+    "tests/v2_tests/test_index_init_logging.py"
+    "tests/v2_tests/test_marqo_cloud_instance_mapping.py"
+    "tests/v2_tests/test_index_manipulation_features.py"
+    "tests/v2_tests/test_index.py"
+    "tests/v2_tests/test_get_indexes.py"
+    "tests/v2_tests/test_hybrid_search.py"
+    "tests/v2_tests/test_logging.py"
+    "tests/v2_tests/test_recommend.py"
+  ];
+
+  dependencies =
+    [ requests packaging pip setuptools urllib3 pydantic requests-mock ];
```

```suggestion
  dependencies = [
    packaging
    pydantic
    requests
    typing-extensions
    urllib3
  ];
```
Please respect `install_requires`.
https://github.com/marqo-ai/py-marqo/blob/fbf4d7b54f9e20917a6f37a17a54755b20657965/setup.py#L8-L15

---

@natsukium thanks for your guidance. Just stuck on this one as of today: tests seem to be failing without `requests-mock`... even when i tried loading into `optional-dependencies` (I got the list of test dependencies from [here](https://github.com/marqo-ai/py-marqo/blob/mainline/requirements.txt) and the optional-dependencies pattern sampling from a few other packages)

I've stripped down this list to as minimal as I can get it to build. What's the suggested/standard course of action here: should I just disable tests that can't run without requests-mock?

Thanks for your guidance in advance